ddtrace/_trace/span.py

```diff
@@ -631,11 +634,28 @@ def set_link(
             )
         )
 
-    def _set_link_object(self, link: SpanLink) -> None:
-        # We will be changing this behavior to allow certain kinds of span
-        # links to coexist in the _links list even if they have the same
-        # span_id. For now, we are basically reimplementing the old
-        # dictionary-like behavior.
+    def _add_span_pointer(
+        self,
+        pointer_kind: str,
+        pointer_direction: _SpanPointerDirection,
+        pointer_hash: str,
+        extra_attributes: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        # This is a Private API for now.
+
+        self._set_link_object(
+            _SpanPointer(
+                pointer_kind=pointer_kind,
+                pointer_direction=pointer_direction,
+                pointer_hash=pointer_hash,
+                extra_attributes=extra_attributes,
+            )
+        )
+
+    def _set_link_object(self, link: Union[SpanLink, _SpanPointer]) -> None:
```

Comment:
I'm a bit worried about this method name being confusing since it's only applicable for span pointers. Could we change it to something like `_set_pointer_link` or `append_pointer_link`?

Reply:
well, it takes *either* a span link or a span pointer. how about `_set_link_or_pointer` instead?